src/linked_list.py

```diff
@@ -16,6 +22,9 @@ def push(self, contents):
 
     def pop(self):
         """Remove and return the current head node."""
+        if not self.head_node:
+            print("Linked list is already empty")
```

Comment:
See what exception the basic Python list raises when you try to pop an empty list, and use that.